### PR TITLE
Sticky header

### PR DESCRIPTION
**Pull request by andygout**
cc @i-like-robots

### CODE REVIEW

File: handle-scroll.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
+module.exports = function () {
+
+	const header = document.getElementsByClassName('n-header')[0];
```
Comment (i-like-robots) [inline handle-scroll.js]: `getElementById` I reckon

File: handle-scroll.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+	}
+
+	let animationFrame;
+
+	function startLoop () {
+		animationFrame = window.requestAnimationFrame(function () {
+			if (window.pageYOffset > headerBottom) {
+				stickyNav.setAttribute('aria-hidden', 'false');
+			} else {
+				stickyNav.setAttribute('aria-hidden', 'true');
```
Comment (i-like-robots) [inline handle-scroll.js]: Should sticky nav _always_ be hidden as the usual header will always be accessible?

File: header_new/_header-sticky.scss [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+		table-layout: fixed;
+		width: 100%;
+	}
+
+	.n-header-top__column--left,
+	.n-header-top__column--right {
+		display: table-cell;
+		width: 0;
+		white-space: nowrap;
+	}
```
Comment (i-like-robots) [inline header_new/_header-sticky.scss]: this is all duplicated isn't it?

**Comment by ironsidevsquincy** [comment]
Maybe look into using the `IntersectionObserver` - should have better performance than the scroll event, and it's now in the polyfill service - https://github.com/Financial-Times/polyfill-service/pull/684

Should mean you don't have to worry about this shifting about on the page either

**Comment by ironsidevsquincy** [comment]
Also, have you thought about using `position: sticky` if it's available, and only falling back to a JS option if not?

**Comment by i-like-robots** [comment]
Jazzy animations mean no sticky, sadly

**Comment by i-like-robots** [comment]
re. `IntersectionObserver` because it's still a working draft proposal and Chrome has the only (incomplete) implementation is it even safe to use?

File: handle-scroll.js (hunk 1 above)
Comment (andygout) [inline handle-scroll.js]: Okay - will add the `id` here: https://github.com/Financial-Times/n-header-footer/blob/master/header_new/template.html#L9

File: header_new/_header-sticky.scss (hunk 3 above)
Comment (andygout) [inline header_new/_header-sticky.scss]: Yes - will delete.

**Comment by ironsidevsquincy** [comment]
Is it incomplete in Chrome? It's working well on the front page for lazy loading the images. I would say it's definitely worth using

**Comment by i-like-robots** [comment]
I had a flick through the draft issues and Chromium issue and it seemed like there were lots of questions still remaining - by incomplete I meant not finished, not stable rather than missing features 😶 

If it's been working well on the homepage I see no reason why we can't use it here - though I would think the `requestAnimationFrame` loop will be smoother than the polyfill for users not using Chrome.

**Comment by ironsidevsquincy** [comment]
I see. I'd hope the polyfill would do something clever around using `requestAnimationFrame` if available/only using a single scroll listener, but tbh I haven't looked into it...

File: handle-scroll.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+	let pxFromPageTop = 0;
+
+	function debounce (callback, wait) {
+		let timeout;
+
+		return function () {
+			clearTimeout(timeout);
+			timeout = setTimeout(callback, wait);
+		};
+	}
```
Comment (i-like-robots) [inline handle-scroll.js]: As we're doing elsewhere, perhaps:

```js
const debounce = require('o-viewport/src/utils').debounce;
```

File: handle-scroll.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+	let animationFrame;
+
+	function startLoop () {
+		animationFrame = window.requestAnimationFrame(function () {
+			if (window.pageYOffset > (window.innerHeight / 2)) {
```
Comment (i-like-robots) [inline handle-scroll.js]: maybe comment that this is completely arbitrary

File: header_new/_header-sticky.scss [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+				transform: translateY(0);
+			}
+		}
+	}
+
+	.n-header-top__wrapper {
+		display: table;
+		table-layout: fixed;
+		width: 100%;
+	}
```
Comment (i-like-robots) [inline header_new/_header-sticky.scss]: duplicated

File: header_new/_header-sticky.scss [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+
+	.n-header-top__link--myft {
+		font-size: 0;
+	}
+
+	.n-header-nav__subscribe-button {
+		@include oButtons($theme: 'standout');
+		text-transform: initial;
+		margin-right: 20px;
+	}
```
Comment (i-like-robots) [inline header_new/_header-sticky.scss]: I think we already have these base styles here:

https://github.com/Financial-Times/n-header-footer/blob/master/header_new/_header-nav.scss#L81-L83

Don't worry about the text transform, I already had this conversation - just stick to the o-button style.

**Comment by i-like-robots** [comment]
🚢

**Pull request merged**